### PR TITLE
Using local state for operators (Rx, Ry, Rz, Hadamard) 

### PR DESCRIPTION
**Pull request by Mboky**
Hi,

Thank you again for your response on the issue [Operators (Rx, Ry, Rz, Hadamard) do not support non-Spin Hilbert spaces. #5](https://github.com/netket/netket_fidelity/issues/5) I implemented the code according to your suggestion and it works as expected. I deleted my previous merge request and fixed the typo in the code and reformatted using Black

Kind regards,

Mohammed Boky

### CODE REVIEW

File: netket_fidelity/operator/singlequbit_gates.py [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
-                """The Hilbert space used by Rx must be a `Spin` space.
-
-                This limitation could be lifted by 'fixing' the method
-                `get_conn_and_mels` to work with arbitrary hilbert spaces, which
-                should be relatively straightforward to do, but we have not done so
-                yet.
-                """
-            )
         super().__init__(hi)
+        self._local_states = hi.local_states
```
Comment (PhilipVinc) [inline netket_fidelity/operator/singlequbit_gates.py:13]: Cast it to jax array so jax does not have to do it at every function call

```suggestion
        self._local_states = jnp.asarray(hi.local_states)
```

Comment (PhilipVinc) [reply netket_fidelity/operator/singlequbit_gates.py:13]: should also be done for the other arguments

File: netket_fidelity/operator/singlequbit_gates.py [hunk 2]
```diff
@@ -96,34 +82,30 @@ def to_local_operator(self):
...
+@partial(jax.vmap, in_axes=(0, None, None, [None, None]), out_axes=(0, 0))
+def get_conns_and_mels_Rx(sigma, idx, angle, local_states):
     assert sigma.ndim == 1
 
-    conns = jnp.tile(sigma, (2, 1))
-    conns = conns.at[1, idx].set(-conns.at[1, idx].get())
+    state_0 = jnp.asarray(local_states[0], dtype=sigma.dtype)
+    state_1 = jnp.asarray(local_states[1], dtype=sigma.dtype)
+
+    cons = jnp.tile(sigma, (2, 1))
```
Comment (PhilipVinc) [inline netket_fidelity/operator/singlequbit_gates.py:92]: minor, but could you keep this named `conns` everywhere? it's the convention we use throughout netket
```suggestion
    conns = jnp.tile(sigma, (2, 1))
```

Comment (PhilipVinc) [reply netket_fidelity/operator/singlequbit_gates.py:92]: please fix other occurrences as well.

**Review by PhilipVinc — COMMENTED** [review]
Thanks. I edited a few things to make all CI work decently now.

Your edits make sense and should work, but could you add a test to `test/test_operator.py` that checks that the connected elements belong to the Hilbert space correctly? We will work a bit more on this code in the future and having a test ensures we don't break it without noticing.

otherwise it's good to be merged

**Comment by Mboky** [comment]
Thank you for the feedback. I will include some tests, fix the inconsistencies and implement the approvements you mention above. For the coming commit should I also commit the changes Black makes in other files or will this clutter the pull request too much?

**Comment by PhilipVinc** [comment]
It's fine don't worry. 
The repo is small for now

**Comment by PhilipVinc** [comment]
it would still be nice to have it.

**Comment by Mboky** [comment]
Hi I am redoing it all in one commit right now. I was gone for a bit too long. I'm currently writing the tests.

**Pull request closed without merging**